### DISCUSSION TITLE
How do i dynamically hide and show permission by Soft Delete

### DISCUSSION THREAD

**Discussion by bayukartiko** [spatie/laravel-permission#2315] [Q&A]
hello, in my case I want to make a list of all permission from permission table and I have done add custom permission model like this:
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\SoftDeletes;
use Spatie\Permission\Models\Permission as PermissionsRole;

class Permission extends PermissionsRole
{
    use SoftDeletes;
    
    public static function aliasName(){
        return "Permission";
    }
}
```

And I have created the "deleted_at" column inside permission table
```
Schema::create($tableNames['permissions'], function (Blueprint $table) {
    $table->bigIncrements('id');
    $table->string('name');             // For MySQL 8.0 use string('name', 125);
    $table->string('guard_name')->default('web'); // For MySQL 8.0 use string('guard_name', 125);
    $table->timestamps();
    $table->softDeletes(); // <-- here!

    $table->unique(['name', 'guard_name']);
});
```

But when I set `(for example "access all resource" )` column `deleted_at` value to not null.
The problem is when I check user auth()->user()->hasPermissionTo("access all resource") always returns true (because I have set it before and now the permission are soft deleted). how do I make `return false` when specific permission is soft deleted from the database? 

Thank you...

**Comment by bayukartiko** [comment]
why I don't use the `->revokePermissionTo("access all resource")`? that's because my website has some tricky user features like "standard", "professional" and many more. That's my soft delete to do for permission.

when I soft delete that permission, the user or role that has that permission will not be revoked, it still has access to that permission but in this case, permission is soft deleted then act like `"you don't have this permission"` even though the permission still exists in user/role.

and when I restore that soft-deleted permission, the user/role will automatically have access to that permission without `revoking` or `giving` to them manually.

**Comment by parallels999** [comment]
You can handle that on your own implementation
